internal/settings/gandi.go

```diff
@@ -0,0 +1,199 @@
+package settings
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/qdm12/ddns-updater/internal/constants"
+	"github.com/qdm12/ddns-updater/internal/models"
+	"github.com/qdm12/ddns-updater/internal/regex"
+)
+
+const DefaultTTL = 3600
+
+type gandi struct {
+	domain    string
+	host      string
+	ttl       int
+	ipVersion models.IPVersion
+	dnsLookup bool
```

Comment:
No need for the `dnsLookup` option, only `cloudflare` needs it actually because it proxies domains

```suggestion
```